### PR TITLE
My Profile Front End Automation

### PR DESCRIPTION
**Pull request by manas-NTech**
Segregate front End Automation.

### CODE REVIEW

File: go1percent/page-objects/My_Profile/myProfilePage.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        OverallRank: "body > app-root:nth-child(1) > div:nth-child(1) > app-main:nth-child(2) > section:nth-child(1) > main:nth-child(2) > div:nth-child(2) > div:nth-child(1) > div:nth-child(1) > app-my-profile:nth-child(2) > div:nth-child(1) > div:nth-child(1) > div:nth-child(1) > div:nth-child(1) > div:nth-child(2) > div:nth-child(2) > h4:nth-child(2)",
+        MonthlyScore: "div[class='d-flex flex-column align-items-center points-section p-1'] h4[class='mt-2']",
+        RewardExist: "img[alt='reward']",
+    
+        Badge: {
+            locateStrategy: 'xpath',
+            selector: "//body/app-root/div/app-main/section[@class='g-sidenav-show']/main[@class='main-content position-relative max-height-vh-100 h-100 border-radius-lg']/div[@class='container-fluid pb-4']/div[@class='row']/div[@class='col-12']/app-my-profile/div[@class='row board-pos']/div[@class='col-lg-8']/div[@class='card mt-2 card-gap']/div[@class='row badge-card p-3']/div[@class='d-flex justify-content-start align-items-center no-scroll-badges']/div[1]/div[1]/div[1]//*[name()='svg']"
```
Comment (Sparsh79) [inline go1percent/page-objects/My_Profile/myProfilePage.js:52]: Please use a better locator strategy. Do not use absolute xpaths.

Comment (manas-NTech) [reply go1percent/page-objects/My_Profile/myProfilePage.js:52]: Updated all the absolute xpath to css selector.

File: go1percent/test/myProfile_FE/myProfile.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    it('View the profile picture along with the monthly rank', function (browser) {
+        myProfile
+            .pause(5000)
+            .assert.elementPresent('@ProfilePicExist', 'Profile Pic exist')
+            .assert.elementPresent('@ViewMonthlyRank', 'Monthly Rank exist');
+    });
+    it('Change their profile picture by clicking on update profile page', function (browser) {
+        myProfile
+
+            .pause(7000)
```
Comment (deepansh257) [inline go1percent/test/myProfile_FE/myProfile.js:56]: Too many pause commands. Please remove and use wait commands

File: go1percent/test/myProfile_FE/myProfile.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        myProfile
+            .pause(15000)
+            .assert.elementPresent('@RewardExist', 'All redeemed reward for the user displyed')
+            .assert.elementPresent('@ViewRewardBtn', 'No reward redeemed for the user');
+
+
+    });
+    it('View the badges and the count in the badges section', function (browser) {
+
+        browser.elements('xpath', "//div[@class='w-20']", function (result) {
```
Comment (deepansh257) [inline go1percent/test/myProfile_FE/myProfile.js:97]: Do not put any locator in test file. Add all of them in page object and call from there.

**Review by deepansh257 — CHANGES_REQUESTED** [review]

**Review by Sparsh79 — APPROVED** [review]

File: go1percent/test/myProfile_FE/myProfile.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+            .assert.urlContains("my-profile", 'URL contains my-profile')
+
+    });
+    afterEach(function (browser) {
+        browser.end();
+    });
+
+    it("No Badges Earned' message when no badges were earned to their profile", function (browser) {
+
+        browser.elements('xpath', "//div[@class='w-20']", function (result) {
```
Comment (deepansh257) [inline go1percent/test/myProfile_FE/myProfile.js:38]: put all the locators from test file in page object file

**Review by deepansh257 — CHANGES_REQUESTED** [review]

File: go1percent/nightwatch.conf.js [hunk 5]
```diff
@@ -15,10 +15,11 @@ module.exports = {
   // An array of folders (excluding subfolders) where your tests are located;
   // if this is not specified, the test source must be passed as the second argument to the test runner.
   src_folders: ['./test'],
+  skip_testcases_on_fail: false,
 
 
   // See https://nightwatchjs.org/guide/concepts/page-object-model.html
-  page_objects_path: ['page-objects', 'page-objects/Techhub_AdminUser_FE'],
```
Comment (Ritika-gupta20) [inline go1percent/nightwatch.conf.js:21]: Why have you deleted this, please add for your module and do not delete other module code from anywhere

**Review by Ritika-gupta20 — CHANGES_REQUESTED** [review]
Please do the changes in config.js, do not remove code of other modules

**Comment by Ritika-gupta20** [comment]
Please resolve conflicts

File: go1percent/nightwatch.conf.js (hunk 5 above)
Comment (manas-NTech) [reply go1percent/nightwatch.conf.js]: Added my page-objects path.

Comment (Ritika-gupta20) [inline go1percent/nightwatch.conf.js]: please don't remove others code and replace it with yours

**Review by Ritika-gupta20 — APPROVED** [review]

**Review by deepansh257 — APPROVED** [review]

**Pull request merged**